### DISCUSSION TITLE
iptc:keywords can't have spaces in them.

### DISCUSSION THREAD

**Discussion by d-air1** [photoprism/photoprism#2344] [General]
For example lets say that a picture has the keyword "New York". When indexing my photos this gets turned into two keywords 
1. "new"
2. "york"

In exiftool the keywords section looks to be a comma separated list. For example:
keyword: "New York, Empire State Building, night, holiday" 

I am not manually tagging my photos. I use a different piece of software to automatically tag them according to what the image contains. I use digikam whenever I need to edit. These tags show up just fine in digikam, but the issue as described above happens in photoprism. 
I have read various issues on photoprism's github. From what I understand I know that it removes trailing and leading whitespace, it allows dashes unless they are trailing, and that it doesn't allow words under three characters. 

Anyways, my question is can i make photoprism process my tags "as is". Like I mentioned before. I'm already using software to automatically tag my images, so I know that they are how I have intended them to be.

I just wanted to ask here before filing a bug report.

**Comment by lastzero** [comment]
You are right: The keywords are actual words, with which other words e.g. contained in a search query get compared against. This implements a simple, vendor/database independent text search. Obviously this is not the right data structure and approach for structured tags and compound terms our backend currently knows nothing about.

This is because a) we have a small team and b) services like Solr are great at handling/searching such text data so there is no need to reinvent the wheel. It's just that not everyone wants to run an enterprise architecture at home and most users are happy with what we provide today.

Meaning such functionality is not available, but planned. Likely shipping with our upcoming version for professional users first. Nested tags with different types fall into the same category of advanced organization features, mainly requested by professional users such as scientists who need them for proper categorization. Generally, I guess you would need stats and some kind of language model to decide what results match for example for New York. York is a different city indeed. There are heaps of locations with many more words in their name, and different languages likely have different rules... Google uses a top k search with 3 words max, from what I know. Search results for queries with more words rapidly decrease in quality and depth. We could implement the same and compete not only with Google Photos but also their core product.

**Reply by d-air1** [reply]
I see thanks for answering.
